src/BTVNanoCommissioning/workflows/ctag_DY_valid_sf.py

```diff
@@ -134,19 +138,47 @@ def process(self, events):
                 -1,
             )
         if isRealData:
-
             output["sumw"] = len(events)
+            if self.isJERC:
+                if "un" in dataset:
+                    jecname = (
+                        dataset[dataset.find("un") + 6]
+                        if dataset[dataset.find("un") + 2].isdigit()
+                        else dataset[dataset.find("un") + 2]
+                    )
+                elif "data" in dataset:
+                    jecname = dataset[dataset.find("data") + 4]
```

Comment:
It would be better to have same naming scheme like would be nice, `$dataset_$Run_$campaign`
`SingleMuon_Run2017B-UL2017_MiniAODv2_NanoAODv9-v1`
Then I can just simplify to `the dataset[dataset.find("un") + 6]` version

Reply:
I agree, we should then also add in the readme that a fixed naming scheme needs to be used for the samples